Frontend/student-hub/src/pages/CoursePage.js

```diff
@@ -240,16 +296,44 @@ function CoursePage() {
                         <Card>
                         <CardContent>
                             <Typography variant="h5">{post.title}</Typography>
+                            {post.mediaPath && (
+                            <>
+                                {/\.(mp4|mov|avi)$/i.test(post.mediaPath) ? (
+                                <video controls style={{ maxWidth: '100%' }}>
+                                    <source src={`http://localhost:8080/${post.mediaPath}`} type="video/mp4" />
+                                    Your browser does not support the video tag.
+                                </video>
+                                ) : (
+                                <img
+                                    src={`http://localhost:8080/${post.mediaPath}`}
+                                    alt="Post media"
+                                    style={{ maxWidth: '100%' }}
+                                />
+                                )}
+                            </>
+                            )}
                             <Typography>{post.content}</Typography>
                             {post.User && (
-                                <Typography>By: {post.User.name}</Typography>
+                                <Typography>By: {post.User?.name}</Typography>
                             )}
-                            <Typography>{post.Comments.length} Comments</Typography>
-                            <Typography>{post.Likes.length} Likes</Typography>
+                            <Typography>{post.Comments?.length} Comments</Typography>
+                            <Typography>{post.Likes?.length} Likes</Typography>
                             </CardContent>
                             <CardActions>
-                                <IconButton onClick={() => handleLike(post.id)}>
-                                    <ThumbUpIcon />
+                            <IconButton onClick={() => handleLike(post.id)}>
+                                    <Select
+                                        value={selectedEmoji[post.id] || ''}
+                                        onChange={(e) => setSelectedEmoji({ ...selectedEmoji, [post.id]: e.target.value })}
+                                        displayEmpty
+                                        inputProps={{ 'aria-label': 'Select Emoji' }}
+                                    >
+                                         <MenuItem value=""><span role="img" aria-label="default emoji">👍🏻</span></MenuItem>
+                                        {emojis.map((emoji, index) => (
+                                            <MenuItem key={index} value={emoji.character}>
```

Comment:
it's usually not preferred to use the array index as the `key` here! there's an explanation in the React docs here: https://react.dev/learn/rendering-lists#why-does-react-need-keys